### ISSUE TITLE
versions 3.0.0-beta6 and 3.0.0-rc.1 both fail with react-stylegudist

### ISSUE THREAD

**Issue by matthew-dean** [reactjs/react-docgen#303]
Since updating Styleguidist, I've been getting these errors:
```
Warning: Cannot parse src/common/presentation/elements/Anchor/Anchor.js: RangeError: Maximum call stack size exceeded

It usually means that react-docgen does not understand your source code, try to file an issue here:
https://github.com/reactjs/react-docgen/issues

Warning: Cannot parse src/common/presentation/elements/Avatar/Avatar.js: RangeError: Maximum call stack size exceeded

It usually means that react-docgen does not understand your source code, try to file an issue here:
https://github.com/reactjs/react-docgen/issues
```

However, react-styleguidist works with the dependency of `3.0.0-beta12`. Beta 6 AND RC 1 both fail. See a similar error here: https://github.com/reactjs/react-docgen/issues/249

**Comment by danez** [comment]
please add an example

**Comment by jquense** [comment]
The loop seems to be isolated to function components, but the below isn't working with rc.1 with all exported components resolver
```js
export default function MainContent({
  size = 'medium',
  centerText,
  className,
  ...props
}) {
  return (
    <ResponsiveContext.Consumer>
      {layout => (
        <div
          className={classNames(
            className,
            styles[layout],
            styles[size],
            centerText && styles.centerText,
          )}
          {...props}
        />
      )}
    </ResponsiveContext.Consumer>
  );
}
```

**Comment by jquense** [comment]
looks like the problem is in `IsStatelessComponent`, perhaps a babel type changed or something that isn't ever breaking the recursion?

**Comment by jquense** [comment]
it appears that https://github.com/reactjs/react-docgen/blob/master/src/utils/isStatelessComponent.js#L149 is always hit (scopes are never equal) and it just vists this node forever, which seems like an odd recast behavior?

**Comment by jquense** [comment]
maybe: https://github.com/benjamn/recast/issues/374 ?

**Comment by danez** [comment]
hmmm I'm not able to reproduce this. I tried the example on latest master and on rc.1 and both parse successfully.

This is what I tested:

```js
import React from 'react';

export default function Spinner({ className, ...props }) {
  return <svg viewBox="0 0 34 34" {...props} />;
}
```

**Comment by danez** [comment]
We released 3.0.0-rc.2, it has the latest version of recast and babel and some fixes. Let me know if that makes it any better.
